BuildResidentialScheduleFile/resources/schedules.rb

```diff
@@ -776,15 +797,15 @@ def get_ev_battery_schedule(away_schedule, hours_driven_per_year)
         first_half_driving = (actual_driving_time / 2.0).ceil
         second_half_driving = actual_driving_time - first_half_driving
 
-        discharging_schedule += [1] * first_half_driving  # Start driving
-        discharging_schedule += [0] * idle_time           # Idle in the middle
-        discharging_schedule += [1] * second_half_driving # End driving
-        charging_schedule += [0] * activity_minutes
+        discharging_schedule.concat([1] * first_half_driving)  # Start driving
+        discharging_schedule.concat([0] * idle_time)           # Idle in the middle
+        discharging_schedule.concat([1] * second_half_driving) # End driving
+        charging_schedule.concat([0] * activity_minutes)
 
         driving_minutes_used += actual_driving_time
       else
-        charging_schedule += [1] * activity_minutes
-        discharging_schedule += [0] * activity_minutes
+        charging_schedule.concat([1] * activity_minutes)
+        discharging_schedule.concat([0] * activity_minutes)
```

Comment:
Specifically for EVs, this code was also resulting in some slowdown since `+=` creates new arrays and all of this is happening within a loop.